app/src/ui/repository-settings/git-config.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react'
+import { DialogContent } from '../dialog'
+import { TextBox } from '../lib/text-box'
+import { Row } from '../lib/row'
+
+interface IGitConfigProps {
+  readonly gitConfigLocation: GitConfigLocation
+  readonly name: string
+  readonly email: string
+  readonly globalName: string
+  readonly globalEmail: string
+
+  readonly onGitConfigLocationChanged: (value: GitConfigLocation) => void
+  readonly onNameChanged: (name: string) => void
+  readonly onEmailChanged: (email: string) => void
+}
+
+interface IGitConfigState {
+  readonly gitConfigLocation: GitConfigLocation
+}
+
+export enum GitConfigLocation {
+  Global = 'Global',
+  Local = 'Local',
+}
+
+/** A view for creating or modifying the repository's gitignore file */
+export class GitConfig extends React.Component<
+  IGitConfigProps,
+  IGitConfigState
+> {
+  public constructor(props: IGitConfigProps) {
+    super(props)
+
+    this.state = {
+      gitConfigLocation: this.props.gitConfigLocation,
+    }
+  }
+  private onGitConfigLocationChanged = (
+    event: React.FormEvent<HTMLInputElement>
+  ) => {
+    const value = event.currentTarget.value as GitConfigLocation
+
+    this.setState({ gitConfigLocation: value })
+    this.props.onGitConfigLocationChanged(value)
+  }
+
+  public render() {
+    return (
+      <DialogContent>
+        <div className="advanced-section">
+          <h2>For this repository I wish to</h2>
+          <div className="radio-component">
+            <input
+              type="radio"
+              id={GitConfigLocation.Global}
+              value={GitConfigLocation.Global}
+              checked={
+                this.state.gitConfigLocation === GitConfigLocation.Global
+              }
+              onChange={this.onGitConfigLocationChanged}
+            />
+            <label htmlFor={GitConfigLocation.Global}>
+              Use my global Git config
```

Comment:
Thanks! I think a missing piece here is letting them know what their global git config is, so they can say, "ok cool that's not what I want for this repo, I'm going to override it with this other thing." It's just an additional measure of certainty.

Reply:
Not shown above but imagine the above screenshot with the Name and Email textbooks disabled when the Global radio is selected

Reply:
🤔 I'd like to hear @ampinsk's thoughts, but my initial reaction is that I'd prefer to display the global config as a static element, and then have the local config fields be blank when you select "Use my local config."

Reply:
Hey @say25 and @billygriffin, how about this: 

<img width="837" alt="Screen Shot 2020-04-10 at 3 08 06 PM" src="https://user-images.githubusercontent.com/10404068/79026436-b3eb8680-7b3d-11ea-856a-caaf1c97f673.png">

So this would show the global config in disabled inputs when that option was selected, and when the Local option is selected, it would enable the fields and let you input your new info. How does this look to both of you?

Reply:
Looks good to me. I can make the change